resolwe/storage/tests/test_listener.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-docstring
+from unittest.mock import MagicMock, patch
+
+from resolwe.flow.managers.listener import ExecutorListener
+from resolwe.flow.managers.protocol import ExecutorProtocol
+from resolwe.storage.models import FileStorage, ReferencedPath, StorageLocation
+from resolwe.test import TestCase
+
+
+class ListenerTest(TestCase):
+    def setUp(self):
+        self.listener = ExecutorListener()
+        self.file_storage = FileStorage()
+        self.file_storage.save()
+        self.storage_location = StorageLocation(
+            file_storage=self.file_storage, connector_name="GCS", status="OK"
+        )
+        self.storage_location.save()
+        path = ReferencedPath.objects.create(
+            path="test.me", md5="md5", crc32c="crc", awss3etag="aws"
+        )
+        self.storage_location.files.add(path)
+
+    def test_download_finished(self):
+        storage_location = StorageLocation(
+            file_storage=self.file_storage, connector_name="local"
+        )
+        obj = {
+            "command": ExecutorProtocol.DOWNLOAD_FINISHED,
+            "data_id": -1,
+            "storage_location_id": -2,
+        }
+        send_wrapper = MagicMock()
+        send_reply = MagicMock()
+        async_to_sync = MagicMock(return_value=send_wrapper)
+        with patch(
+            "resolwe.storage.models.FileStorage.default_storage_location",
+            self.storage_location,
+        ):
+            with patch("resolwe.flow.managers.listener.async_to_sync", async_to_sync):
+                with patch(
+                    "resolwe.flow.managers.listener.ExecutorListener._send_reply",
+                    send_reply,
+                ):
+                    self.listener.handle_download_finished(obj)
+
+        async_to_sync.assert_called_once_with(send_reply)
+        send_wrapper.assert_called_once_with(
+            {"command": "download_finished", "data_id": -1, "storage_location_id": -2,},
+            {"result": "ER"},
+        )
+
+        storage_location.save()
```

Comment:
Why do you need this save?

Reply:
Since it was actually not saved before. Now the code is a little cleaner in this regard.